### PR TITLE
Centralize versioning for resource manager libraries

### PR DESCRIPTION
**Pull request by JimSuplizio**
This PR contains the following changes
1. Addition of resource manager libraries to eng/versionining/version_client.txt
2. Modification of the pom file version scanner to add the additional resource manager groupId (com.azure.resourcemanager) used to wildcard a groupId for the enforcer plugin.
3. Modification of all resource-manager libraries to parent off of azure-client-sdk-parent instead of azure-resourcemanager-parent. This includes version tagging for all dependencies and plugins. Note: sdk/resourcemanager/azure-resourcemanager-parent/pom.xml was not deleted as part of this PR. It was left for reference purposes and should be deleted as part of [Resource Manager library adjustments for azure-client-sdk-parent parentage](https://github.com/Azure/azure-sdk-for-java/issues/13249)
4. Addition of missing README.md files for all resource manager libraries.
5. Addition of build plugin configuration for maven-enforcer-plugin in the libraries that required it.
6. Set low min jacoco numbers in the properties.
7. Added all resourcemanager libraries to eng/jacoco-test-coverage

It's worth noting that each and every resource manager library's pom file has the following entries for the build/plugin configuration for maven-compiler-plugin:

`                    
                    <failOnWarning>false</failOnWarning>
`

`                   
                    <showDeprecation>true</showDeprecation>
`

failOnWarning of false was one of the settings inherited from the azure-resourcemanager-parent, azure-client-sdk-parent has this as true. The expectation is that this should be turned off and using azure-client-sdk-parent's setting which is to fail on warning. Right now, turning this off will cause builds to fail as there's a lot of warnings. From what I can tell most of these existed prior to my changes.

### CODE REVIEW

File: sdk/parents/azure-client-sdk-parent/pom.xml [hunk 1]
```diff
@@ -529,6 +530,9 @@
             <excludePackageNames>
               *.impl*:
               *.implementation*:
+              <!-- RMDocs do utils and samples actually do anything -->
+              *.utils:
+              *.samples
```
Comment (JimSuplizio) [inline sdk/parents/azure-client-sdk-parent/pom.xml:535]: @srnagar we'd chatted about this earlier. For right now I've opted to leave both of these entries in here. I'm going to add the "RMDocs" as one of the searchable tags that you'll be looking at as part of [Resource Manager library adjustments for azure-client-sdk-parent parentage](https://github.com/Azure/azure-sdk-for-java/issues/13249)

File: sdk/parents/azure-client-sdk-parent/pom.xml [hunk 2]
```diff
@@ -662,6 +666,10 @@
                 <title>Azure Text Analytics</title>
                 <packages>com.azure.ai.textanalytics*</packages>
               </group>
+              <group>
+                  <title>Azure Management</title>
+                  <packages>com.azure.management*</packages>
```
Comment (mitchdenny) [inline sdk/parents/azure-client-sdk-parent/pom.xml:671]: Is this correct?

Comment (JimSuplizio) [reply sdk/parents/azure-client-sdk-parent/pom.xml:671]: It's what was originally in their parent so I'm assuming so.

**Pull request merged**